python-venv/cmstools/test/cfs_sessions_rc_test/cfs/session.py

```diff
@@ -0,0 +1,247 @@
+#
+# MIT License
+#
+# (C) Copyright 2025 Hewlett Packard Enterprise Development LP
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+
+"""
+CFS Session related functions
+"""
+
+import requests
+
+from cmstools.lib.api import request, request_and_check_status
+from cmstools.lib.cfs.defs import (CFS_SESSIONS_URL_TEMPLATE, HTTP_OK, HTTP_NO_CONTENT, HTTP_BAD_REQUEST,
+                                   HTTP_NOT_FOUND, SessionGetWithNameContainsResult)
+from cmstools.lib.api.api import API_REQUEST_TIMEOUT, add_api_auth, SYSTEM_CA_CERTS
+from cmstools.test.cfs_sessions_rc_test.log import logger
+from cmstools.test.cfs_sessions_rc_test.defs import CFSRCException
+from cmstools.test.cfs_sessions_rc_test.defs import CFS_VERSIONS_STR
+
+
+def get_next_id(data: dict) -> str | None:
+    """
+    Get the next_id from the data if it exists.
+    """
+    next_obj = data.get("next")
+    if isinstance(next_obj, dict):
+        return next_obj.get("after_id")
+    return None
+
+
+def get_session_data(cfs_session_data: dict | list[dict]) -> list | None:
```

Comment:
This is all coming in from request responses that are decoded JSON, so we can safely type the dicts as `JsonDict`

```suggestion
from cmstools.lib.defs import JsonDict
from cmstools.test.cfs_sessions_rc_test.log import logger
from cmstools.test.cfs_sessions_rc_test.defs import CFSRCException
from cmstools.test.cfs_sessions_rc_test.defs import CFS_VERSIONS_STR


def get_next_id(data: JsonDict) -> str | None:
    """
    Get the next_id from the data if it exists.
    """
    next_obj = data.get("next")
    if isinstance(next_obj, dict):
        return next_obj.get("after_id")
    return None


def get_session_data(cfs_session_data: JsonDict | list[JsonDict]) -> list[JsonDict] | None:
```